### PR TITLE
raft/documentation: clarify progress's subjects.

### PR DESCRIPTION
**Pull request by gyuho**
If I understand correctly, `progress` represents the states of follower. For
me, some comments weren't clear because it was missing the subjects of
`progress`. This adds more clarification on who is doing what. Please let me
know if I misunderstood anything. Thanks,

### CODE REVIEW

File: raft/design.md [hunk 1]
```diff
@@ -37,11 +37,11 @@ receives msgAppResp(rej=true)
                   +--------------------+                                                        
 ```
 
-When in `probe` state, leader sends at most one `replication message` per heartbeat interval. The leader sends `replication message` slowly and probing the actual progress of the follower. A `msgHeartbeatResp` or a `msgAppResp` with reject might trigger the sending of the next `replication message`.
+When a follower is in `probe` state, leader sends at most one `replication message` per heartbeat interval. The leader sends `replication message` slowly and probing the actual progress of the follower. A `msgHeartbeatResp` or a `msgAppResp` with reject might trigger the sending of the next `replication message`.
```
Comment (xiang90) [inline raft/design.md]: When the progress of a follower is in

**Comment by xiang90** [comment]
@gyuho LGTM. Actually, this part is not included in raft paper. We did this for some practical reason since etcd/raft supports pipelining requests. Do you think they are reasonable to you? If they do not make sense to you or you want us to clarify the doc more, I also love to do it. The goal is to make people easy to understand etcd/raft by reading its code and doc.

**Comment by gyuho** [comment]
@xiang90 Just updated the comments as you suggest. I just found out the design.md and it's really helpful to understand the internals of etcd. I only wish we would have more of this for beginners like me. Yes, the doc makes sense to me and we should have more if you have some extra time. Thanks

**Comment by xiang90** [comment]
> have more if you have some extra time

I would love to do this for etcd/raft pkg as a starter. Which part do you think we can improve? Any examples?

**Comment by gyuho** [comment]
If you ask for any example, I think rethinkdb has the best documentation (https://www.rethinkdb.com/docs/architecture/) and it really helps better its user experience. But I personally really like current etcd documentation (it was easy enough for me to set up with 0 background).

And I think for developers, the **high level overview** of architecture is really helpful in general to get people engaged. I have seen something like this from consul (https://www.consul.io/docs/internals/architecture.html).

**Pull request merged**